server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -947,6 +952,11 @@ public void initRestHandlers(Supplier<DiscoveryNodes> nodesInCluster) {
             registerHandler.accept(new RestDeleteSearchPipelineAction());
         }
 
+        // Extensions API
+        if (FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS)) {
+            registerHandler.accept(new RestInitializeExtensionAction(extensionsManager));
```

Comment:
Yes because of the chicken-egg problem described in the issue and the PR description, the plan is to remove loading extensions during bootstrap completely and making it dynamic. This would be helpful for hot-swap as well in the future.